UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -949,25 +952,77 @@ protected virtual async Task BuildAnimationSamplers(GLTFAnimation animation, int
 			{
 				var time = input.AsFloats[i];
 
-				var values = getConvertedValues(output, i);
+				float[] values = null;
+				float[] inTangents = null;
+				float[] outTangents = null;
+				if (mode == InterpolationType.CUBICSPLINE)
+				{
+					// For cubic spline, the output will contain 3 values per keyframe; inTangent, dataPoint, and outTangent.
+					// https://github.com/KhronosGroup/glTF/blob/master/specification/2.0/README.md#appendix-c-spline-interpolation
+
+					var cubicIndex = i * 3;
+					inTangents = getConvertedValues(output, cubicIndex);
+					values = getConvertedValues(output, cubicIndex + 1);
+					outTangents = getConvertedValues(output, cubicIndex + 2);
+				}
+				else
+				{
+					// For other interpolation types, the output will only contain one value per keyframe
+					values = getConvertedValues(output, i);
+				}
 
 				for (var ci = 0; ci < channelCount; ++ci)
 				{
-					keyframes[ci][i] = new Keyframe(time, values[ci]);
+					if (mode == InterpolationType.CUBICSPLINE)
+					{
+						keyframes[ci][i] = new Keyframe(time, values[ci], inTangents[ci], outTangents[ci]);
+					}
+					else
+					{
+						keyframes[ci][i] = new Keyframe(time, values[ci]);
+					}
 				}
 			}
 
 			for (var ci = 0; ci < channelCount; ++ci)
 			{
-				// set interpolcation for each keyframe
-				SetCurveMode(keyframes[ci], mode);
 				// copy all key frames data to animation curve and add it to the clip
 				AnimationCurve curve = new AnimationCurve();
 				curve.keys = keyframes[ci];
+
+				// For cubic spline interpolation, the inTangents and outTangents are already explicitly defined.
+				// For the rest, set them appropriately.
+				if (mode != InterpolationType.CUBICSPLINE)
```

Comment:
same, suggest being explicit with the type you need this code for. make it easy to add new animation types to the standard without causing going into unexpected paths

Reply:
This doesn't seem like the kind of thing that is likely to change in the standard.  I could see it changing via an extension, but then the person adding support for the standard would naturally end up fixing things here.